src/OrchardCore.Modules/OrchardCore.Autoroute/Handlers/AutoroutePartHandler.cs

```diff
@@ -112,6 +112,13 @@ public override async Task UpdatedAsync(UpdateContentContext context, AutorouteP
             // Compute the Path only if it's empty
             if (!String.IsNullOrWhiteSpace(part.Path))
```

Comment:
Remove this because it prevents from having a unique path for each of them.

Reply:
I think this is ok because otherwise it would try and generate the path every time, which would not work if the path has been set manually, and not with a pattern.

I have not yet added a `slugify` check here like we talked about, but I would rather check with `slugify` as well, and then make a unique path based on the data provided, than generating with a guid. (unless you have reason that is a bad idea?). That way the path (or alias) provided, is as close to what was requested by either the api or content item generated from code as possible.

There is another option.
We talked about validation.
We could make the ContentApiController run the display manager on the content item, and return any validation errors without creating the content item.

That might not be a bad idea.

I saw for example that it is possible to make a `BlogPost` without all the required fields (taxonomy and tags) so while the `BlogPost` creates, it is not a valid content item

Reply:
Ok here you need to take in consideration that from an API Call the event will be triggered. If you pass in the part.Path a string for example "Foo Bar". This string will fail to generate a unique path from that first "if" statement.